### ISSUE TITLE
hitting space while reading in a room with no new messages = infinite loop?

### ISSUE THREAD

**Issue by gordon-morehouse** [dgets/DOCshell#52]
This behavior is best experienced in Lobby> on the test system currently. I don't know if it has anything to do with the deleted messages in Lobby> or not. However, once I start reading the room by hitting space at the room prompt, I see the sole non-deleted message, "msg #0 (11 remaining)" at the read prompt. I then hit space again. Currently on the test system, there are no more (undeleted) messages in Lobby>.

WHAT ACTUALLY HAPPENS
------

1. I see a bunch of debugging output, mainly to do with skipping the deleted messages.
2. If I hit space again, I see the same message again, but the read prompt now says "msg &#35;10 (1 remaining)".
3. If I hit space yet again, I see some debug output *and* a room prompt, *and then* the same message again as the system appears to have automatically started reading the room again?
```
[Lobby> msg #10 (1 remaining)] Read cmd ->  
End of scanSub() main loop
tmpPtr: 11	inc: 1	fuggit: false
In main scanSub() loop	tmpPtr: 11
ptr: 11	base.last_msg: 11
Got exception name: dispMsgException	Msg: Invalid message slot
Num: 1
In main scanSub() loop	tmpPtr: 12
tp: 12	inc: 1	mBase.last_msg: 11	mBase.code: undefined	mBase.is_open: true
Hit last message, returning 1
Exception: verifyBoundsException	Msg: Last message pointer	#: 1
ecode: 1
Got topegrplobby in user.cursub
Lobby> Entered scanSub(); forward = true  user.cursub: topegrplobby
sBoard.code: topegrplobby
Opened: topegrplobby allegedly . . .
sBoard.scan_ptr = 0
mBase.first_msg = 10
mBase.total_msgs = 2
mBase.last_msg = 11
Inc: 1	based on 'forward'
In main scanSub() loop	tmpPtr: 0
ptr: 0	base.last_msg: 11
Wed, 21 Jan 2015 06:21:58 -0800 from REDACTED
  I'm in the process of working out the last of the bugs here.  As soon as the
reverse reading and 's'kip functionality is implemented this base will be
active and utilized by the new DOC shell.

[Lobby> msg #0 (11 remaining)] Read cmd -> Stop
```

WHAT I EXPECT TO HAPPEN
------

Upon hitting space at the read prompt on the last readable message in a room, I should be dropped to the room prompt.

**Comment by ntwitch** [comment]
Should be fixed as of 22b50be817d9a990a76711c066388d15986026c8

**Comment by dgets** [comment]
See also: https://github.com/dgets/DOCshell/pull/77#issuecomment-77310046
